src/middlewares/navigate.middleware.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import { get } from 'lodash/fp';
+import history from 'utils/history.utils';
+
+import type { Middleware } from 'types/redux.types';
+
+const redirectMiddleware: Middleware = () => {
+  return next => action => {
+    if (!get('meta.redirect', action)) {
+      return next(action);
+    }
+
+    history.replace('/login');
```

Comment:
this should be more generic. Think about creating specific navigate actions that you can tell where you want to go.

for example this action:
`navigateTo('/login')` which returns the following action `{ type: 'NAVIGATE', payload: { url } }`

then the middleware will read this from the action.payload.url and navigate to where you need.

This way in the future if you want to navigate after some sort of action, or from a component click handler (when you can't use <Link/>) it will be very easy to do.

Reply:
💯 , changed.